src/Chainweb/Miner.hs

```diff
@@ -125,6 +141,37 @@ toMinerData = MinerData . encodeToByteString
 {-# INLINABLE toMinerData  #-}
 
 -- | Convert from Chainweb 'MinerData' to Pact Miner
+--
 fromMinerData :: MonadThrow m => MinerData -> m Miner
 fromMinerData = decodeStrictOrThrow' . _minerData
 {-# INLINABLE fromMinerData #-}
+
+-- | Rewards table mapping 3-month periods to their rewards
+-- according to the calculated exponential decay over 120 year period
+--
+readRewards
+    :: HasChainGraph v
+    => v -> HashMap BlockHeight ParsedDecimal
+readRewards v =
+    case CSV.decode CSV.NoHeader (toS rawMinerRewards) of
+      Left e -> error
+        $ "cannot construct miner reward map: "
+        <> sshow e
+      Right vs -> HM.fromList
+        . V.toList
+        . V.map formatRow
+        $ vs
+  where
+    formatRow :: (Word64, Double) -> (BlockHeight, ParsedDecimal)
+    formatRow (!a,!b) =
+      let
+        !n = v ^. chainGraph . to (int . size)
+        !m = fromRational $ toRational b
+      in (BlockHeight a, ParsedDecimal $ m / n)
+{-# INLINE readRewards #-}
+
+-- | Read in the reward csv via TH for deployment purposes
+--
+rawMinerRewards :: ByteString
+rawMinerRewards = $(embedFile "rewards/miner_rewards.csv")
+{-# INLINE rawMinerRewards #-}
```

Comment:
I don't think this should be inlined. How big is the file? The body of this function contains the whole spread sheet, doesn't it?

So, instead, I'd rather add an explicit `{-# NOINLINE #-}` pragma, in order to make sure that we don't copy the data to each call site in the binaries.

Reply:
Good point

Reply:
I'll do that, thanks. I was being overzealous.